### ISSUE TITLE
Auth0UserDetails missing getter and non conforming to spring security

### ISSUE THREAD

**Issue by joedunne** [auth0/auth0-spring-security-mvc#5]
There is no public getter method for the extraInfo object. Please add.

public Map<String, Object> getExtraInfo() {
    return extraInfo;
}

Also the following methods are supposed to return true according to the Spring API:

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

**Comment by arcseldon** [comment]
@joedunne - Thank you for this. Please can you provide a link to the Spring spec stating above - we can update accordingly.  What information did you wish to retrieve from extraInfo?

**Comment by jeffsheets** [comment]
@arcseldon Thanks for looking into this! I have been following this issue and can comment on the need for getExtraInfo()

We have userMetadata and appMetadata sections in the Auth0 profile for each user. We need to access this data for the loggedIn User. The appMetadata is stored in the extraInfo of the Auth0UserDetails, but we cannot access it because there is no public getter.

Also the userMetadata is not available in the extraInfo. So for a workaround for now we had to override Auth0UserDetails to add the public getExtraInfo() and a new property to hold for a getUserMetadata().

A general comment, maybe for a different issue, but it would be useful to be able to easily override the Auth0UserDetails implementation with our own class. Maybe by using a Auth0UserDetailsProvider or similar for the creation of that object, or calling the Auth0UserDetailsService to create the object. Just as a possible idea.

**Comment by jeffsheets** [comment]
Also here are the docs for the spring security API to help with your other question:
http://docs.spring.io/spring-security/site/docs/current/apidocs/org/springframework/security/core/userdetails/UserDetails.html

```boolean isAccountNonLocked()
Indicates whether the user is locked or unlocked. A locked user cannot be authenticated.```
```boolean isAccountNonExpired()
Indicates whether the user's account has expired. An expired account cannot be authenticated.```
```boolean isCredentialsNonExpired()
Indicates whether the user's credentials (password) has expired. Expired credentials prevent authentication.```

**Comment by joedunne** [comment]
I need the appMetaData and userMetaData from the extraInfo.  I simply need this method:

public Map getExtraInfo() {
return extraInfo;
}

**Comment by arcseldon** [comment]
@jeffsheets / @joedunne - thanks both for the added info. Shall update with next version.

@jeffsheets - please can you share the code for your overridden version of the UserDetails object? Appreciate it is pretty simple, just wished to have something to compare against.

**Comment by arcseldon** [comment]
The required changes have been merged as PR - awaiting publication to Maven - shall be in next release.

**Comment by arcseldon** [comment]
available with version 1.2.0
